### PR TITLE
Add integration tests and release automation

### PR DESCRIPTION
**Pull request by yahavi**
Depends on: https://github.com/jfrog/jfrog-testing-infra/pull/1

Add integration tests according to [Bamboo wired tests framework](https://developer.atlassian.com/server/framework/atlassian-sdk/run-wired-tests-with-the-plugin-test-console/).
A detailed description of the test can be found in the [CONTRIBUTING.md](https://github.com/yahavi/bamboo-artifactory-plugin/blob/tests-pr/CONTRIBUTING.md).

To allow overriding dummy Artifactory credentials configured in the Bamboo configuration UI, I added the 2 missing following Bamboo properties support:
```
artifactory.override.resolver.url
artifactory.override.deployer.url
```

An example of successful running in the JFrog Pipelines can be found [here](https://ecosysjfrog.jfrog.io/ui/pipelines/myPipelines/default/test_bamboo?branch=tests-pr2).

### CODE REVIEW

File: src/test/java/it/org/jfrog/bamboo/utils/Utils.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+     */
+    public static String downloadBuildLog(Logger log, int buildNumber, String planKey) {
+        String url = String.format("%s/download/%s-JOB1/build_logs/%s-JOB1-%s.log", BAMBOO_TEST_URL, planKey, planKey, buildNumber);
+        HttpGet request = new HttpGet(url);
+        request.setHeader(HttpHeaders.AUTHORIZATION, ENCODED_AUTH);
+        try (CloseableHttpClient httpClient = HttpClients.createDefault();
+             CloseableHttpResponse response = httpClient.execute(request)) {
+            HttpEntity entity = response.getEntity();
+            if (entity != null) {
+                return EntityUtils.toString(entity);
```
Comment (eyalbe4) [inline src/test/java/it/org/jfrog/bamboo/utils/Utils.java:89]: I think the enrity should be consumed before returning:
```
EntityUtils.consumeQuietly(response.getEntity());
```

File: src/test/java/it/org/jfrog/bamboo/utils/Utils.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            }
+        } catch (IOException e) {
+            log.error("Couldn't read build log: " + ExceptionUtils.getRootCauseMessage(e));
+        }
+        return "";
+    }
+
```
Comment (eyalbe4) [inline src/test/java/it/org/jfrog/bamboo/utils/Utils.java:96]: redundant new-line here.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
LIKE! 👍

File: src/test/java/it/org/jfrog/bamboo/utils/Utils.java (hunk 1 above)
Comment (yahavi) [reply src/test/java/it/org/jfrog/bamboo/utils/Utils.java:89]: The entity's Inputstream can be read only once, therefore it should be closed in the toString method. See:
https://github.com/apache/httpcomponents-core/blob/rel/v4.4.5/httpcore/src/main/java/org/apache/http/util/EntityUtils.java#L252

**Pull request merged**